google/cloud/bigquery/dbapi/_helpers.py

```diff
@@ -189,12 +188,20 @@ def bigquery_scalar_type(value):
     elif isinstance(value, numbers.Real):
         return "FLOAT64"
     elif isinstance(value, decimal.Decimal):
-        # We check for NUMERIC before BIGNUMERIC in order to support pyarrow < 3.0.
-        scalar_object = pyarrow.scalar(value)
-        if isinstance(scalar_object, pyarrow.Decimal128Scalar):
+        vtuple = value.as_tuple()
+        # NUMERIC values have precision of 38 (number of digits) and scale of 9 (number
+        # of fractional digits), and their max absolute value must be strictly smaller
+        # than 1.0E+29.
+        # https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#decimal_types
+        if (
+            len(vtuple.digits) <= 38  # max precision: 38
+            and vtuple.exponent >= -9  # max scale: 9
+            and _NUMERIC_SERVER_MIN <= value <= _NUMERIC_SERVER_MAX
+        ):
             return "NUMERIC"
```

Comment:
The logic differs from the [docs](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#decimal_types), because testing the actual backend response showed that `9.9999999999999999999999999999999999999E+29` is not accepted as a valid NUMERIC value:
```sql
SELECT CAST((SELECT "9.9999999999999999999999999999999999999E+29" AS literal) AS NUMERIC);
```
Even `1.0E+29` is too large, but `9.999...9E+28` works, meaning that there's probably an off-by-one error in the docs for the exponent.